### ISSUE TITLE
Not working as of at least 3.12.0b3

### ISSUE THREAD

**Issue by GanerCodes** [chilaxan/fishhook#5]
```
import fishhook

@hook(int)
def __add__(self, other):
  print(other)
  return orig(self, other)
```

↓

```
Traceback (most recent call last):
  File "/tmp/what.py", line 1, in <module>
    import fishhook
  File "/home/ganer/.local/lib/python3.12/site-packages/fishhook/__init__.py", line 18, in <module>
    from .fishhook import *
  File "/home/ganer/.local/lib/python3.12/site-packages/fishhook/fishhook.py", line 139, in <module>
    patch_object()
  File "/home/ganer/.local/lib/python3.12/site-packages/fishhook/fishhook.py", line 113, in patch_object
    tp_dict_offset = find_offset(int_mem, id(getdict(int)))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ganer/.local/lib/python3.12/site-packages/fishhook/fishhook.py", line 53, in find_offset
    return [*mem].index(val)
           ^^^^^^^^^^^^^^^^^
ValueError: 140659297419968 is not in list
```

**Comment by chilaxan** [comment]
i have a fix for the ValueError, but hooks in general seem to not work on 3.12+ as is so i will need to make more changes to the code

**Comment by chilaxan** [comment]
just pushed an update (0.2.9) that should fix fishhook on 3.12+, please lmk if you have any other issues
